compiler/rustc_llvm/llvm-wrapper/CoverageMappingWrapper.cpp

```diff
@@ -58,10 +60,38 @@ fromRust(LLVMRustCounterMappingRegionKind Kind) {
     return coverage::CounterMappingRegion::GapRegion;
   case LLVMRustCounterMappingRegionKind::BranchRegion:
     return coverage::CounterMappingRegion::BranchRegion;
+  case LLVMRustCounterMappingRegionKind::MCDCDecisionRegion:
+    return coverage::CounterMappingRegion::MCDCDecisionRegion;
+  case LLVMRustCounterMappingRegionKind::MCDCBranchRegion:
+    return coverage::CounterMappingRegion::MCDCBranchRegion;
   }
   report_fatal_error("Bad LLVMRustCounterMappingRegionKind!");
 }
 
+// FFI equivalent of struct `llvm::coverage::CounterMappingRegion::MCDCParameters`
+// https://github.com/rust-lang/llvm-project/blob/66a2881a/llvm/include/llvm/ProfileData/Coverage/CoverageMapping.h#L253-L263
+struct LLVMRustMCDCParameters {
+    /// Byte Index of Bitmap Coverage Object for a Decision Region.
+    uint32_t BitmapIdx = 0;
+
+    /// Number of Conditions used for a Decision Region.
+    uint32_t NumConditions = 0;
+
+    /// IDs used to represent a branch region and other branch regions
+    /// evaluated based on True and False branches.
+    uint32_t ID = 0;
+    uint32_t TrueID = 0;
+    uint32_t FalseID = 0;
+};
+
+static coverage::CounterMappingRegion::MCDCParameters
+fromRust(LLVMRustMCDCParameters MCDCParams) {
+  return coverage::CounterMappingRegion::MCDCParameters{
+    MCDCParams.BitmapIdx, MCDCParams.NumConditions,
+    MCDCParams.ID, MCDCParams.TrueID, MCDCParams.FalseID
+  };
+}
+
```

Comment:
This is a good example of how we should be passing data from Rust to C++. I think something like this is what we should be using in #123409.

However, the fields should be declared as `unsigned`, so that they exactly match the type declarations that LLVM uses. And the corresponding struct on the Rust side should use `libc::c_uint` to match.

That makes it easier to verify that the FFI mapping has been done correctly.